packages/ui-components/src/components/windowedlist.ts

```diff
@@ -841,6 +895,11 @@ export class WindowedList<
     if (this._viewModel.windowingActive) {
       this._removeListeners();
     }
+
+    if (!this._scrollbar.isDisposed) {
+      this._scrollbar.dispose();
```

Comment:
Could you elaborate the need to dispose the scrollbar on before-detach message?

Reply:
This was because I was trying to maintain the `Widget` lifecycle myself instead of either:
1. relying on `Layout` to do the work
2. containing the scrollbar within the `WindowedList` class
I tried the first, but it was too messy, the second is really the way to go.